R/stroke.R

```diff
@@ -240,6 +240,70 @@ cross_check_links <- function(best_links) {
   return(links)
 }
 
+#' @noRd
+to_linestring <- function(stroke, segments, nodes) {
```

Comment:
Simplifying `traverse_segments` and the main body of `merge_lines` made this function more complex, but I guess this is how it has to be. 

Also, I had to take this and `traverse_segments` back out from `merge_lines` because the linter would otherwise still raise issues with `merge_lines` still having too large cyclomatic complexity. But I have double-checked and R should pass variables by reference (until these are not modified), so this should not give rise to any unexpected double memory usaget..